concreep-refilled/control.lua

```diff
@@ -55,6 +55,92 @@ function init()
+			surface.create_entity{name="entity-ghost", position={target_x_pos-half_distance_x,target_y_pos+offset}, inner_name="big-electric-pole", force=force, expires=false}
+		end
+		if (target_x_pos < roboport_x_pos and 
+		surface.can_place_entity{name="entity-ghost", position={target_x_pos-half_distance_x,target_y_pos+offset}, inner_name="big-electric-pole", force=force, build_check_type=defines.build_check_type.script_ghost}and 
+		surface.count_entities_filtered{area={{target_x_pos-half_distance_x-10,target_y_pos+offset-10},{target_x_pos-half_distance_x+10,target_y_pos+offset+10}},name="entity-ghost"} == 0) then
+			surface.create_entity{name="entity-ghost", position={target_x_pos-half_distance_x,target_y_pos+offset}, inner_name="big-electric-pole", force=force, expires=false}
+		end	
+	end
+end
+
+function hypercreep(roboport)
+	debug_print_function_was_called("hypercreep")
+	local roboport_item_count = roboport.logistic_network.get_item_count("roboport")
+	local power_pole_item_count = roboport.logistic_network.get_item_count("big-electric-pole")
+	if (roboport_item_count < 4 or power_pole_item_count < 4)
+		debug_print("Not enough roboports/power poles to hyprecreep")
+		return 
+	end if
+	local offset_power_poles = 3
+	local logistic_diameter = roboport.logistic_cell.logistic_radius * 2
+	for i=1, 8 do
+		local target_x_pos = roboport.position.x
+		local target_y_pos = roboport.position.y
+		if i==1 then
+			target_x_pos = target_x_pos + logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)
+		elseif i==2 then
+			target_x_pos = target_x_pos - logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)
+		elseif i==3 then
+			target_y_pos = target_y_pos + logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)
+		elseif i==4 then
+			target_y_pos = target_y_pos - logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)
+ 		elseif i==5 then
+			target_x_pos = target_x_pos + logistic_diameter
+			target_y_pos = target_y_pos + logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)
+		elseif i==6 then
+			target_x_pos = target_x_pos - logistic_diameter
+			target_y_pos = target_y_pos - logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)
+		elseif i==7 then
+			target_x_pos = target_x_pos + logistic_diameter
+			target_y_pos = target_y_pos - logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)
+		elseif i==8 then
+			target_x_pos = target_x_pos - logistic_diameter
+			target_y_pos = target_y_pos + logistic_diameter
+			hypercreep_builder(target_x_pos,target_y_pos,offset_power_poles,roboport)		 
+		end
+	end	
+end	

```

Comment:
there must be a better way to do this - maybe something like a switch statement? maybe you can help

Reply:
uhm - yeah dont know why i insisted on using a for loop here - i think it was for testing quickly what different patterns would look like but this definitely makes much more sense

Reply:
weirldy enough - your solution does not work as intended - the top left corner of the grid does not get build - cant quite see whats wrong here - maybe you should try the mod out in game as well - with editor extensions its rather nice to debug - doing more testing right now

Reply:
turns out - thats not your code - would have been really wondering there - because it should be identical to mine in terms of functionality - need to check what causes this behaviour